marshmallow/fields.py

```diff
@@ -445,15 +445,17 @@ def _serialize(self, nested_obj, attr, obj):
         if not self.__updated_fields:
             schema._update_fields(obj=nested_obj, many=self.many)
             self.__updated_fields = True
-        ret, errors = schema.dump(nested_obj, many=self.many,
+        try:
+            ret = schema.dump(nested_obj, many=self.many,
                 update_fields=not self.__updated_fields)
-        if isinstance(self.only, basestring):  # self.only is a field name
-            if self.many:
-                return utils.pluck(ret, key=self.only)
-            else:
-                return ret[self.only]
-        if errors:
-            raise ValidationError(errors, data=obj, valid_data=ret)
+        except ValidationError as exc:
+            raise ValidationError(exc.messages, data=obj, valid_data=exc.valid_data)
+        finally:
```

Comment:
This causes a `NameError` to be raised instead of the `ValueError` when using the string version of `only`, because `ret` will always be undefined if an exception is raised. This shouldn't have been wrapped in a `finally` block. 

```py
try:
    raise ValueError()
    a = 1
except ValueError as e:
    raise ValueError()
finally:
    print(a)
# Traceback (most recent call last):
#   line 7, in <module>
# NameError: name 'a' is not defined
```

The should get fixed as part of #800.

Reply:
Good point @deckar01. I wonder how I missed this.

Should I do something about it or just wait for #800?

Reply:
I have a fix in a WIP PR. https://github.com/marshmallow-code/marshmallow/pull/823